python/ray/train/examples/tune_cifar_torch_pbt_example.py

```diff
@@ -58,17 +58,42 @@ def validate_epoch(dataloader, model, loss_fn):
     return {"loss": test_loss}
 
 
+def update_optimizer_config(optimizer, config):
+    for param_group in optimizer.param_groups:
+        for param, val in config.items():
+            param_group[param] = val
+
+
 def train_func(config):
-    epochs = config.pop("epochs", 3)
     model = resnet18()
     model = train.torch.prepare_model(model)
 
     # Create optimizer.
-    optimizer = torch.optim.SGD(
-        model.parameters(),
-        lr=config.get("lr", 0.1),
-        momentum=config.get("momentum", 0.9),
-    )
+    optimizer_config = {
+        "lr": config.get("lr"),
+        "momentum": config.get("momentum"),
+    }
+    optimizer = torch.optim.SGD(model.parameters(), **optimizer_config)
+
+    epoch = 0
+    if session.get_checkpoint():
+        checkpoint_dict = session.get_checkpoint().to_dict()
+
+        # Load in model
+        model_state = checkpoint_dict["model_state_dict"]
+        model.load_state_dict(model_state)
+
+        # Load in optimizer
+        optimizer_state = checkpoint_dict["optimizer_state_dict"]
+        optimizer.load_state_dict(optimizer_state)
+
+        # Optimizer configs (`lr`, `momentum`) are being mutated by PBT and passed in
+        # through config, so we need to update the optimizer loaded from the checkpoint
+        update_optimizer_config(optimizer, optimizer_config)
```

Comment:
I think we just need to also save and load the learning rate scheduler state (which holds the epoch information), along with the optimizer. Then, PBT will perturb the LR, but it will still follow the same schedule. We could show how to do it in this example or another example maybe?